packages/@react-aria/virtualizer/src/Virtualizer.tsx

```diff
@@ -78,9 +78,15 @@ function Virtualizer<T extends object, V>(props: VirtualizerProps<T, V>, ref: Re
 
   // Handle scrolling, and call onLoadMore when nearing the bottom.
   let isLoadingRef = useRef(isLoading);
+  let prevProps = useRef(props);
   useLayoutEffect(() => {
-    isLoadingRef.current = isLoading;
-  }, [isLoading]);
+    // Only update isLoadingRef if props object actually changed,
```

Comment:
The problem is that we set `isLoadingRef.current = true` in `onVisibleRectChange` when we fire `onLoadMore`, but the Virtualizer itself may re-render before `props.isLoading` becomes true (due to `state.setVisibleRect`), or it may never become true. We don't want to reset `isLoadingRef.current` back to the value from props if there was only a local state update, we only want to do so if the prop actually changed. That ensures we don't call `onLoadMore` twice due to the layout effect checking the content size below after the re-render.